drivers/network/netio/netio.c

```diff
@@ -0,0 +1,860 @@
+    {
+        case SOCK_DGRAM:
+            if (Protocol != IPPROTO_UDP)
+            {
+                DbgPrint("SOCK_DGRAM only supports IPPROTO_UDP\n");
+                status = STATUS_INVALID_PARAMETER;
+                goto err_out;
+            }
+            if (Flags != WSK_FLAG_DATAGRAM_SOCKET)
+            {
+                DbgPrint("SOCK_DGRAM flags must be WSK_FLAG_DATAGRAM_SOCKET\n");
+                status = STATUS_INVALID_PARAMETER;
+                goto err_out;
+            }
+            break;
+
+        case SOCK_STREAM:
+            if (Protocol != IPPROTO_TCP)
+            {
+                DbgPrint("SOCK_STREAM only supports IPPROTO_TCP\n");
+                status = STATUS_INVALID_PARAMETER;
+                goto err_out;
+            }
+            if ((Flags != WSK_FLAG_CONNECTION_SOCKET) && (Flags != WSK_FLAG_LISTEN_SOCKET))
+            {
+                DbgPrint("SOCK_STREAM flags must be either WSK_FLAG_CONNECTION_SOCKET or WSK_FLAG_LISTEN_SOCKET\n");
+                status = STATUS_INVALID_PARAMETER;
+                goto err_out;
+            }
+            break;
+
+        case SOCK_RAW:
+            DbgPrint("SOCK_RAW not supported\n");
+            status = STATUS_NOT_SUPPORTED;
+            goto err_out;
+
+        default:
+            status = STATUS_INVALID_PARAMETER;
+            goto err_out;
+    }
+
+    s = ExAllocatePoolWithTag(NonPagedPool, sizeof(*s), 'SOCK');
+    if (s == NULL)
+    {
+        DbgPrint("WskSocket: Out of memory\n");
+        status = STATUS_INSUFFICIENT_RESOURCES;
+        goto err_out;
+    }
+    s->family = AddressFamily;
+    s->type = SocketType;
+    s->proto = Protocol;
+    s->flags = Flags;
+    s->user_context = SocketContext;
+    s->LocalAddressHandle = NULL;
+    s->LocalAddressFile = NULL;
+    s->Flags = 0;
+    s->ListenDispatch = Dispatch;
+    s->RefCount = 1;            /* SocketPut() is in WskCloseSocket */
+    s->ConnectionHandle = NULL;
+
+    switch (SocketType)
+    {
+        case SOCK_DGRAM:
+            s->s.Dispatch = &UdpDispatch;
+            RtlInitUnicodeString(&s->TdiName, L"\\Device\\Udp");
+            break;
+        case SOCK_STREAM:
+            s->s.Dispatch = &TcpDispatch;
+            RtlInitUnicodeString(&s->TdiName, L"\\Device\\Tcp");
+
+            status = TdiOpenConnectionEndpointFile(&s->TdiName, &s->ConnectionHandle, &s->ConnectionFile);
+            if (status != STATUS_SUCCESS)
+            {
+                DbgPrint("Could not open TDI handle, status is %x\n", status);
+                ExFreePoolWithTag(s, 'SOCK');
+                goto err_out;
+            }
+            if (Flags == WSK_FLAG_LISTEN_SOCKET && s->ListenDispatch == NULL)
+            {
+                DbgPrint("Warning: no callbacks given for listen socket\n");
+            }
+            break;
+
+        default:
+            DbgPrint("Socket type not yet supported\n");
+            /* A little bit later this probably crashes */
```

Comment:
Then maybe you should fail the request instead ;)

Reply:
Ok returning STATUS_NOT_SUPPORTED in that case now.